crates/symbolicator-proguard/tests/integration/proguard.rs

```diff
@@ -209,3 +209,695 @@ async fn test_resolving_inline() {
+                module: "io.sentry.samples.MainActivity$InnerClass".into(),
+                filename: Some("MainActivity.kt".into()),
+                lineno: 20,
+                index: 3,
+                ..Default::default()
+            },
+            JvmFrame {
+                function: "whoops2".into(),
+                abs_path: None,
+                module: "io.sentry.samples.MainActivity$AnotherInnerClass".into(),
+                filename: None,
+                lineno: 26,
+                index: 4,
+                ..Default::default()
+            },
+            JvmFrame {
+                function: "whoops3".into(),
+                abs_path: Some("MainActivity.kt".into()),
+                module: "io.sentry.samples.MainActivity$AdditionalInnerClass".into(),
+                filename: Some("MainActivity.kt".into()),
+                lineno: 32,
+                index: 5,
+                ..Default::default()
+            },
+            JvmFrame {
+                function: "whoops4".into(),
+                abs_path: Some("SourceFile".into()),
+                module: "io.sentry.samples.MainActivity$OneMoreInnerClass".into(),
+                filename: Some("SourceFile".into()),
+                lineno: 38,
+                index: 6,
+                ..Default::default()
+            },
+        ],
+    }];
+
+    let request = SymbolicateJvmStacktraces {
+        scope: Scope::Global,
+        sources: Arc::new([source]),
+        exceptions: vec![exception.clone()],
+        stacktraces,
+        modules: vec![JvmModule {
+            uuid: debug_id,
+            r#type: JvmModuleType::Source,
+        }],
+        apply_source_context: true,
+        release_package: None,
+    };
+
+    let CompletedJvmSymbolicationResponse {
+        stacktraces,
+        exceptions,
+        ..
+    } = symbolication.symbolicate_jvm(request).await;
+
+    let exc = &exceptions[0];
+    let frames = &stacktraces[0].frames;
+
+    assert_eq!(exc.ty, "RuntimeException");
+    assert_eq!(exc.module, "io.sentry.samples");
+
+    assert_eq!(frames[0].function, "otherMethod");
+    assert_eq!(frames[0].module, "OtherActivity");
+    assert_eq!(frames[0].lineno, 100);
+    assert!(frames[0].context_line.is_none());
+    assert!(frames[0].pre_context.is_empty());
+    assert!(frames[0].post_context.is_empty());
+
+    assert_eq!(frames[1].function, "differentMethod");
+    assert_eq!(frames[1].module, "DifferentActivity");
+    assert_eq!(frames[1].lineno, 200);
+    assert!(frames[1].context_line.is_none());
+    assert!(frames[1].pre_context.is_empty());
+    assert!(frames[1].post_context.is_empty());
+
+    assert_eq!(frames[2].function, "onCreate");
+    assert_eq!(frames[2].module, "io.sentry.samples.MainActivity");
+    assert_eq!(frames[2].lineno, 11);
+    assert_eq!(
+        frames[2].context_line,
+        Some("        InnerClass().whoops()".into())
+    );
+    assert_eq!(
+        frames[2].pre_context,
```

Comment:
using a snapshot test for these might be a better idea

Reply:
Wish I'd thought of that 😅